client/ethereum.go

```diff
@@ -265,6 +304,28 @@ func (e *EthereumClient) WaitForTransaction(transactionHash common.Hash) error {
 			confirmations++
 			confirmationLog.Msg("Transaction confirmed, waiting on confirmations")
 
+			if confirmations >= minConfirmations {
+				confirmationLog.Msg("Minimum confirmations met")
+				return err
+			} else {
+				confirmationLog.Msg("Waiting on minimum confirmations")
+			}
+		case <-ticker.C:
+			// FIXME: inefficient polling just to debug volume tests batch deployment
+			minConfirmations := e.Network.Config().MinimumConfirmations
+			confirmationLog := log.Debug().Str("Network", e.Network.Config().Name).
+				Str("Tx Hash", transactionHash.Hex()).
+				Int("Minimum Confirmations", minConfirmations).
+				Int("Total Confirmations", confirmations)
+			isConfirmed, err := e.isTxConfirmed(transactionHash)
+			if err != nil {
+				return err
+			} else if !isConfirmed {
+				confirmationLog.Msg("Transaction still pending, waiting for confirmation")
+				continue
+			}
+			confirmations++
+			confirmationLog.Msg("Transaction confirmed, waiting on confirmations")
```

Comment:
Just polling, I think it's out of scope, but no new header was received in the previous case branch, so we were always waiting for timeout.
Make it for being able to run x100 batch deployments, works fine.